### PR TITLE
Document our security stance

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: docs/user/manual.adoc [hunk 1]
```diff
@@ -516,6 +516,20 @@ See https://github.com/rust-analyzer/rust-project.json-example for a small examp
 
 You can set `RA_LOG` environmental variable to `rust_analyzer=info` to inspect how rust-analyzer handles config and project loading.
 
+== Security
+
+At the moment, rust-analyzer assumes that all code is trusted.
+Here is a **no-exhaustive** list of ways to make rust-analyzer execute arbitrary code:
```
Comment (bjorn3) [inline docs/user/manual.adoc:522]:
```suggestion
Here is a **non-exhaustive** list of ways to make rust-analyzer execute arbitrary code:
```

File: docs/user/manual.adoc [hunk 2]
```diff
@@ -516,6 +516,20 @@ See https://github.com/rust-analyzer/rust-project.json-example for a small examp
...
+
+* proc macros and build scripts are executed by default
+* `.cargo/config` can override `rustc` with an arbitrary executable
+* VS Code plugin reads configuration from project directory, and that can be used to override paths to various executables, like `rustfmt` or `rust-analyzer` itself.
+* rust-analyzer's syntax trees library uses a lot of `unsafe` and hasn't been properly audited for memory safety.
+
+rust-analyzer itself doesn't access the network.
+VS Code plugin doesn't access the network unless the nightly channel is selected in the settings.
```
Comment (lnicola) [inline docs/user/manual.adoc:530]:
```suggestion
The VS Code plugin doesn't access the network unless the nightly channel is selected in the settings.
```

File: docs/user/manual.adoc [hunk 3]
```diff
@@ -516,6 +516,20 @@ See https://github.com/rust-analyzer/rust-project.json-example for a small examp
...
+In that case, the plugin uses GitHub API to check for and download updates.
```
Comment (lnicola) [inline docs/user/manual.adoc:531]:
```suggestion
In that case, the plugin uses the GitHub API to check for and download updates.
```

**Comment by matklad** [comment]
cc https://github.com/microsoft/vscode/issues/106488

**Pull request merged**